src/afrolabs/components/http.clj

```diff
@@ -24,6 +23,17 @@
               (filter #(<= 0 % 255))
               count)))))
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;; We want middleware to be applied in order.
+;; To be first in the middleware, means you will receive the request first.
+;; To be last in the middleware, means you will receive the request last, and the response first.
+
+(defprotocol IRingMiddlewareProvider
+  (get-middleware [_] "Returns a collection of ring-style middleware."))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
```

Comment:
define a new interface with which the http component can retrieve middleware from other components